parsers/cxx/cxx_scope.c

```diff
@@ -240,15 +236,11 @@ CXXToken * cxxScopeTakeTop(void)
 			g_pScope->iCount > 0,
 			"When popping as scope there must be a scope to pop"
 		);
+	CXX_DEBUG_PRINT("Popped scope: '%s'",cxxScopeGetFullName() ?: "");
 
 	CXXToken * t = cxxTokenChainTakeLast(g_pScope);
 	g_bScopeNameDirty = true;
 
-#ifdef CXX_DO_DEBUGGING
-	const char * szScopeName = cxxScopeGetFullName();
-
-	CXX_DEBUG_PRINT("Popped scope: '%s'",szScopeName ? szScopeName : "");
-#endif
 	return t;
 }
 
```

Comment:
@pragmaware, do you think this change (moving CXX_DEBUG_PRINT before calling cxxTokenChainTakeLast) is good?

Reply:
Well... yet it's probably a bit clearer to the reader. The popped scope is really the one that was there before.
I suggest we apply this one and your non-GNU-restricted version of the previous diff.